tempodb/backend/block_meta.go

```diff
@@ -48,14 +48,19 @@ func NewBlockMeta(tenantID string, blockID uuid.UUID, version string, encoding E
 // ObjectAdded updates the block meta appropriately based on information about an added record
 //  start/end are unix epoch seconds
 func (b *BlockMeta) ObjectAdded(id []byte, start uint32, end uint32) {
-	startTime := time.Unix(int64(start), 0)
-	endTime := time.Unix(int64(end), 0)
 
-	if b.StartTime.IsZero() || startTime.Before(b.StartTime) {
-		b.StartTime = startTime
+	if start > 0 {
```

Comment:
is this a bug in the current implementation?
add a quick test?

Reply:
> it gets the timestamps from the spans while iterating the TraceIterator, so it can't create an input meta with timestamps

my concern here is that we constrain setting the min/max block times based on:
https://github.com/grafana/tempo/blob/main/modules/storage/config.go#L40

if compaction just uses the span start/stop times then the compacted blocks time ranges will start exceeding this constraint.

Reply:
Good catch, those were definitely not being preserved.   Ok updated `TraceIterator` to include the trace start/end times, and for `encoding.CompleteBlock` to take and honor them and the block-level start/end times from the wal block.